test/boost/token_metadata_test.cc

```diff
@@ -189,3 +189,59 @@ SEASTAR_THREAD_TEST_CASE(test_pending_endpoints_for_replace_with_replicas) {
     BOOST_REQUIRE_EQUAL(token_metadata.pending_endpoints_for(dht::token::from_int64(11), ks_name),
         inet_address_vector_topology_change{});
 }
+
+SEASTAR_THREAD_TEST_CASE(test_endpoints_for_reading_when_bootstrap_with_replicas) {
```

Comment:
A test for replace and leaving would be great as well, but can be left for follow-up.